### PR TITLE
Fix GCC dereference warning

### PR DESCRIPTION
**Pull request by jamiebramwell**
A downstream customer noticed a build failure on GCC 8.3.1 due to a null pointer dereference warning when building in Release mode.

This also adds release builds to our azure CI pipelines.

### CODE REVIEW

**Comment by jamiebramwell** [comment]
ping @junkudo

File: src/serac/physics/integrators/wrapper_integrator.cpp [hunk 1]
```diff
@@ -26,10 +27,14 @@ void LinearToNonlinearFormIntegrator::AssembleElementGrad(const mfem::FiniteElem
...
-  elmat.SetSize(trial_el.GetDof() * trial_el.GetDim(), el.GetDof() * el.GetDim());
-  elmat = 0.;
+  mfem::FiniteElement const* trial_el = trial_fes_.FEColl()->FiniteElementForGeometry(Tr.GetGeometryType());
+
+  if (trial_el) {
+    elmat.SetSize(trial_el->GetDof() * trial_el->GetDim(), el.GetDof() * el.GetDim());
+    elmat = 0.;
+  } else {
+    SLIC_ERROR("Returned trial element is null.");
+  }
```
Comment (white238) [inline src/serac/physics/integrators/wrapper_integrator.cpp:37]:
```suggestion
  SLIC_ERROR_IF(!trial_el, "Returned trial element is null.");

  elmat.SetSize(trial_el->GetDof() * trial_el->GetDim(), el.GetDof() * el.GetDim());
  elmat = 0.;
```

**Review by white238 — APPROVED** [review]

File: src/serac/physics/integrators/wrapper_integrator.cpp (hunk 1 above)
Comment (jamiebramwell) [reply src/serac/physics/integrators/wrapper_integrator.cpp:37]: Sadly, this does not stop the GCC warning.

File: src/serac/numerics/quadrature_data.hpp [hunk 2]
```diff
@@ -139,6 +139,11 @@ class QuadratureData : public SyncableData, public QuadratureDataImpl<T, Quadrat
    */
   QuadratureData(mfem::Mesh& mesh, const int p, const bool alloc = true);
 
+// Turn off null dereference errors for GCC
```
Comment (jamiebramwell) [inline src/serac/numerics/quadrature_data.hpp]:
```suggestion
// Turn off null dereference warnings for GCC
```

File: src/serac/numerics/quadrature_data.hpp [hunk 3]
```diff
@@ -139,6 +139,11 @@ class QuadratureData : public SyncableData, public QuadratureDataImpl<T, Quadrat
    */
   QuadratureData(mfem::Mesh& mesh, const int p, const bool alloc = true);
 
+// Turn off null dereference warnings for GCC
+// TODO add errors if the MFEM types return null pointers
```
Comment (adayton1) [inline src/serac/numerics/quadrature_data.hpp]: It looks like you already added an error below. Is this TODO referring to other cases?

Comment (jamiebramwell) [reply src/serac/numerics/quadrature_data.hpp]: it's more a TODO to fix the underlying warning instead of turning it off. I'll clarify the comment.

Comment (jamiebramwell) [inline src/serac/numerics/quadrature_data.hpp]:
```suggestion
// TODO Fix the underlying possible nullptr dereference warning with the `MaybeOwnedPointer` type.
```

**Review by samuelpmishLLNL — APPROVED** [review]

**Pull request merged**